### PR TITLE
feat: add support to MySQL8.4 for Ubuntu 24.04

### PR DESCRIPTION
**Pull request by Henrrypg**
# Description

This PR aims to run MySQL  on Ubuntu 24.04.

- [x] Test on Ansible HelmChart

### CODE REVIEW

File: roles/mongo_7_0/tasks/main.yml [hunk 1]
```diff
@@ -12,22 +12,75 @@
...
+- name: Add deadsnakes PPA for Python 3.10
+  ansible.builtin.apt_repository:
+    repo: ppa:deadsnakes/ppa
+    state: present
+
+- name: Install venv
+  ansible.builtin.apt:
+    name:
+      - python3.10
+      - python3.10-venv
```
Comment (Ian2012) [inline roles/mongo_7_0/tasks/main.yml:40]: Can't we use the same python version of the host?

Comment (Henrrypg) [reply roles/mongo_7_0/tasks/main.yml:40]: We're getting the following error when use Python3.12 with [mongodb_user](https://docs.ansible.com/ansible/latest/collections/community/mongodb/mongodb_user_module.html) for somewhat reason.

`Unable to connect to database: 'RawConfigParser' object has no attribute 'readfp'"`

It appears to be compatibility issue but the module is updated, the venv where we run the playbook too...

File: molecule/mongo_7_0/molecule.yml
Comment (Ian2012) [inline molecule/mongo_7_0/molecule.yml:1]: I think you need to add the replica set settings to the molecule ansible environment

File: roles/mysql_8_4/defaults/main.yml [hunk 2]
```diff
@@ -11,11 +12,12 @@ mysql_config_file: /etc/mysql/my.cnf
...
 
 MYSQL_8_4_APT_KEYSERVER: keyserver.ubuntu.com
-MYSQL_8_4_APT_KEY: A8D3785C
+# FROM: https://repo.mysql.com/apt/ubuntu/conf/distributions
+MYSQL_8_4_APT_KEY: B7B3B788A8D3785C
 MYSQL_8_4_REPO: deb http://repo.mysql.com/apt//ubuntu/ {{ ansible_distribution_release }} mysql-8.4-lts
 
 MYSQL_8_4_SUPPORTED_DISTRIBUTIONS:
-  - jammy
+  - noble
```
Comment (jfavellar90) [inline roles/mysql_8_4/defaults/main.yml:20]: Is it still possible to install MySQL in Jammy after these changes? Will only Noble be available?

Comment (jfavellar90) [reply roles/mysql_8_4/defaults/main.yml:20]: The thing is that if we still support Jammy as well, we'll need to test it in molecule

File: roles/mysql_8_4/defaults/main.yml [hunk 3]
```diff
@@ -11,11 +12,13 @@ mysql_config_file: /etc/mysql/my.cnf
 mysql_8_4_socket: /var/run/mysqld/mysqld.sock
 
 MYSQL_8_4_APT_KEYSERVER: keyserver.ubuntu.com
-MYSQL_8_4_APT_KEY: A8D3785C
+# FROM: https://repo.mysql.com/apt/ubuntu/conf/distributions
+MYSQL_8_4_APT_KEY: B7B3B788A8D3785C
 MYSQL_8_4_REPO: deb http://repo.mysql.com/apt//ubuntu/ {{ ansible_distribution_release }} mysql-8.4-lts
 
 MYSQL_8_4_SUPPORTED_DISTRIBUTIONS:
   - jammy
```
Comment (Ian2012) [inline roles/mysql_8_4/defaults/main.yml:20]: I think you can safely remove jammy

File: Dockerfile
Comment (Ian2012) [inline Dockerfile:1]: Should those changes be on this PR or should you split them?

**Comment by MoisesGSalas** [comment]
I think we are going to need a table in the readme with supported versions.

I think all the DBs needed for redwood must run on 24.04. I also think we should try to keep compatibility with 22.04 unless is to much of a burden, this means adding the test cases for molecule on the other PRs too

File: Dockerfile
Comment (Henrrypg) [reply Dockerfile:1]: to avoid open a PR just with this change i thought keep it here

File: roles/mysql_8_4/defaults/main.yml (hunk 3 above)
Comment (Henrrypg) [reply roles/mysql_8_4/defaults/main.yml:20]: We'll try to keep the support to Ubuntu22 as long as it's not a problem

**Comment by Henrrypg** [comment]
We can try to close first this one and then i just rebase and mark the matrix in the other PRs @jfavellar90 @MoisesGSalas @Ian2012

**Review by Ian2012 — APPROVED** [review]

**Review by jfavellar90 — APPROVED** [review]

**Review by MoisesGSalas — APPROVED** [review]

**Pull request merged**